be/src/storage/rowset/rowset_writer.cpp

```diff
@@ -226,6 +228,12 @@ Status RowsetWriter::flush_segment(const SegmentPB& segment_pb, butil::IOBuf& da
         }
         RETURN_IF_ERROR(wfile->close());
 
+        if (_context.tablet_schema->keys_type() == KeysType::PRIMARY_KEYS && _context.partial_update_tablet_schema) {
```

Comment:
1. The issue should describe the error phenomenon in the expected behavior and current behavior.
2. Better to add a concise message about the problem and solution in the commit message.